### PR TITLE
Tsocha/support for new aeon in neon 2.2

### PR DESCRIPTION
**Pull request by tsocha**

### CODE REVIEW

**Comment by jennifermyers** [comment]
@tyler-nervana @Neuroschemata can you review this? We'd like to release neon 2.2 today if possible.

**Comment by wei-v-wang** [comment]
This is to address https://github.com/NervanaSystems/private-neon/issues/1001

File: speech/train.py [hunk 1]
```diff
@@ -96,42 +88,8 @@ def data_transform(dl):
...
-                    audio=feats_config,
-                    transcription=transcripts_config,
-                    manifest_filename=dev_manifest,
-                    macrobatch_size=be.bsz,
-                    minibatch_size=be.bsz)
-train = DataLoader(backend=be, config=train_cfg_dict)
-train = data_transform(train)
-dev = DataLoader(backend=be, config=dev_cfg_dict)
-dev = data_transform(dev)
+train = make_loader(train_manifest, alphabet, nbands, max_tscrpt_len, backend_obj=be)
```
Comment (tyler-nervana) [inline speech/train.py]: Can you make `max_utt_len` an argument here and pass it into `common_config`?

File: speech/data/ingest_librispeech.py [hunk 2]
```diff
@@ -13,28 +13,33 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ----------------------------------------------------------------------------
+
+import numpy as np
 import os
 import logging
 import glob
 import fnmatch
+import json
```
Comment (tyler-nervana) [inline speech/data/ingest_librispeech.py]: Unused import

File: speech/data/ingest_librispeech.py [hunk 3]
```diff
@@ -13,28 +13,33 @@
...
 
+from neon.data.aeon_shim import AeonDataLoader
+from neon.data.dataloaderadapter import DataLoaderAdapter
```
Comment (tyler-nervana) [inline speech/data/ingest_librispeech.py]: Unused import

File: speech/data/ingest_librispeech.py [hunk 4]
```diff
@@ -97,6 +102,40 @@ def librispeech_flac_filename(filestr):
     logger.info("Writing manifest file to {}".format(manifest_file))
     return write_manifest(manifest_file, audio_files, txt_files)
 
+def common_config(manifest_file, batch_size, alphabet, nbands, max_tscrpt_len):
+
+    audio_config = {"type": "audio",
+                            "sample_freq_hz": 16000,
```
Comment (tyler-nervana) [inline speech/data/ingest_librispeech.py]: Align `"sample_freq_hz"` and below with `"type"`.

File: speech/data/ingest_librispeech.py [hunk 5]
```diff
@@ -97,6 +102,40 @@ def librispeech_flac_filename(filestr):
...
+                            "max_duration": "30 seconds",
+                            "frame_length": "25 milliseconds",
+                            "frame_stride": "10 milliseconds",
+                            "feature_type": "mfsc",
+                            "emit_length": True,
+                            "num_filters": nbands}
+
+    transcription_config = {"type": "char_map",
+                                      "alphabet": alphabet,
```
Comment (tyler-nervana) [inline speech/data/ingest_librispeech.py]: Alignment here too

**Review by tyler-nervana — CHANGES_REQUESTED** [review]

**Review by tyler-nervana — APPROVED** [review]

**Pull request merged**